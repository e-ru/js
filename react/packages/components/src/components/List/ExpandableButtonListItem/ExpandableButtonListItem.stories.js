import React from "react";
import { storiesOf } from "@storybook/react";

import HomeIcon from "@material-ui/icons/Home";
import ExpandableButtonListItem from "./ExpandableButtonListItem";

storiesOf("ExpandableButtonListItem", module)
  .add("closed", () => (
    <ExpandableButtonListItem
      open={false}
      title="main"
      icon={<HomeIcon />}
      onClickHandler={() => console.log("onClickHandler clicked")}
    />
  ))
  .add("open", () => (
    <ExpandableButtonListItem
      open
      title="main"
      icon={<HomeIcon />}
      onClickHandler={() => console.log("onClickHandler clicked")}
    />
  ));
