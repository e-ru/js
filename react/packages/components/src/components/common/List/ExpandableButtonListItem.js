import React from "react";
import PropTypes from "prop-types";

import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import ExpandLess from "@material-ui/icons/ExpandLess";
import ExpandMore from "@material-ui/icons/ExpandMore";

import { EXPANDABLE_LIST_ITEM_CLASS } from "../../../constants/ui";

const ExpandableButtonListItem = ({ title, open, icon, handleClick }) => {
  return (
    <ListItem className={EXPANDABLE_LIST_ITEM_CLASS} button onClick={handleClick}>
      <ListItemIcon>{icon}</ListItemIcon>
      <ListItemText primary={title} />
      {open ? <ExpandLess /> : <ExpandMore />}
    </ListItem>
  );
};

ExpandableButtonListItem.propTypes = {
  title: PropTypes.string.isRequired,
  open: PropTypes.bool.isRequired,
  icon: PropTypes.object.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default ExpandableButtonListItem;
