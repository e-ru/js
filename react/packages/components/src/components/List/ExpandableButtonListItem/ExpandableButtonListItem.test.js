import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import HomeIcon from "@material-ui/icons/Home";
import ExpandableButtonListItem from "./ExpandableButtonListItem";

configure({ adapter: new Adapter() });

function setup(open) {
  const props = {
    title: "test title",
    open,
    icon: <HomeIcon />,
    handleClick: jest.fn(),
  };

  const wrapper = shallow(<ExpandableButtonListItem {...props} />);

  return {
    wrapper,
  };
}

describe("<ExpandableButtonListItem />", () => {
  test("renders", () => {
    const { wrapper } = setup(true);
    expect(wrapper).toMatchSnapshot();
  });
});
