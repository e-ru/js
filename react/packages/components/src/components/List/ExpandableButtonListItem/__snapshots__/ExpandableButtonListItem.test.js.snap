// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpandableButtonListItem /> renders 1`] = `ShallowWrapper {}`;
